File: reward/domain/reward.go
package domain

type Target struct {
	//具体什么业务
	Biz   string
	BizId int64
	//打赏的名称
	BizName string
	//打赏给谁
	Uid int64
}

type Reward struct {
	Id     int64
	Uid    int64
	Target Target
	Amt    int64
	Status RewardStatus
}
type CodeURL struct {
	URL string
	Rid int64
}

func (r RewardStatus) AsUint8() uint8 {
	return uint8(r)
}

// Completed 判断是否该订单已经结束
func (r Reward) Completed() bool {
	return r.Status == RewardStatusFailed || r.Status == RewardStatusPayed
}

type RewardStatus uint8

const (
	RewardStatusUnknown = iota
	RewardStatusInit
	RewardStatusPayed
	RewardStatusFailed
)

File: frontEnd/webook-frontend/src/api/reward.ts
import { get, post } from './http'

// 打赏相关接口类型定义
export enum RewardStatus {
  Unknown = 0,
  Init = 1,
  Payed = 2,
  Failed = 3
}

export interface PreRewardRequest {
  biz: string
  biz_id: number
  biz_name: string
  target_uid: number // 被打赏的人
  uid: number // 打赏的人
  amt: number // 打赏金额
}

export interface PreRewardResponse {
  code_url: string
  rid: number
}

export interface GetRewardRequest {
  rid: number
  uid: number
}

export interface GetRewardResponse {
  status: RewardStatus
}

// 打赏相关API
export const rewardApi = {
  // 预打赏
  preReward: (data: PreRewardRequest) => {
    return post<PreRewardResponse>('/reward/pre', data)
  },
  
  // 获取打赏状态
  getReward: (params: GetRewardRequest) => {
    return get<GetRewardResponse>('/reward/get', params)
  }
}
